frontend-public/src/HOC/AuthenticationGuard.js

```diff
@@ -1,79 +1,43 @@
 import React, { Component } from "react";
 import { bindActionCreators } from "redux";
 import { connect } from "react-redux";
-import Keycloak from "keycloak-js";
 import hoistNonReactStatics from "hoist-non-react-statics";
-import { isAuthenticated, getKeycloak } from "@/selectors/authenticationSelectors";
-import { authenticateUser, storeKeycloakData } from "@/actions/authenticationActions";
-import { KEYCLOAK } from "@/constants/environment";
+import { isAuthenticated } from "@/selectors/authenticationSelectors";
 import NullScreen from "@/components/common/NullScreen";
-import Loading from "@/components/common/Loading";
+import { getUserInfoFromToken } from "@/actionCreators/authenticationActionCreator";
 
 /**
  * @constant authenticationGuard - a Higher Order Component Thats checks for user authorization and returns the App component if the user is Authenticated.
  */
 
-export const AuthenticationGuard = (WrappedComponent) => {
-  /**
-   * Initializes the KeyCloak client and enables
-   * redirects directly to IDIR login page.
-   *
-   * The method uses async/awaits instead of promises
-   * because we were facing troubles with promise resolving
-   * and changing state.
-   *
-   */
+export const AuthenticationGuard = (isPublic) => (WrappedComponent) => {
   class authenticationGuard extends Component {
     componentDidMount() {
-      this.keycloakInit();
-    }
-
-    async keycloakInit() {
-      // Initialize client
-      const keycloak = Keycloak(KEYCLOAK);
-      await keycloak.init();
+      const token = localStorage.getItem("jwt");
 
-      // Prompt for login using IDIR if not authenticated
-      if (!keycloak.authenticated) {
-        await keycloak.login({
-          idpHint: KEYCLOAK.idpHint,
-        });
+      if (token && !this.props.isAuthenticated) {
+        this.props.getUserInfoFromToken(token);
       }
-
-      // Fetch user info and roles and store them in local storage
-      const userInfo = await keycloak.loadUserInfo();
-      localStorage.setItem("jwt", keycloak.token);
-      this.props.storeKeycloakData(keycloak);
-      this.props.authenticateUser(userInfo);
     }
 
     render() {
-      if (this.props.keycloak) {
-        if (this.props.isAuthenticated) {
-          return <WrappedComponent {...this.props} />;
-        }
-        // Add this block back in once there are user permissions to check against, ie user logs in with BCeID but permissions have not been updated on keycloak.
-        // if (!this.props.isAuthenticated) {
-        //   return <NullScreen type="unauthorized" />;
-        // }
-        return <Loading />;
+      if (this.props.isAuthenticated || isPublic) {
```

Comment:
What's the purpose of `isPublic`?

Reply:
I needed this authGuard to serve two purposes, only return the `WrappedComponent` if the user is authenticated, and also check for authentication at the highest level, that being App.js. 

So instead of creating two guards that do essentially the same thing, I added an `isPublic` param that is able to authenticate the user (if a token exists) and also renders the `WrappedComponent` because it's a public route, (ie the home page).

Reply:
Okay. That makes sense. I was getting the impression that the entire app was being set to public, circumventing the need to be authenticated. The idea of having lower-level components requiring authentication makes sense. Thanks for clarifying